packages/PHPStanStaticTypeMapper/TypeMapper/ArrayTypeMapper.php

```diff
@@ -93,9 +92,6 @@ public function mapToPHPStanPhpDocTypeNode(Type $type): TypeNode
         return new SpacingAwareArrayTypeNode($itemTypeNode);
     }
 
-    /**
-     * @param ArrayType $type
-     */
     public function mapToPhpParserNode(Type $type, ?string $kind = null): ?Node
     {
         return new Name('array');
```

Comment:
How does PHPStorm handles the type removal? Could you confirm with a gif or something that autocomplete in IDE still works?

Reply:
before this PR

![grafik](https://user-images.githubusercontent.com/120441/123523333-3caefd80-d6c3-11eb-8f81-f7dcce83797a.png)

after this change

![grafik](https://user-images.githubusercontent.com/120441/123523338-433d7500-d6c3-11eb-9df4-5b6aa2122cdd.png)

Reply:
Thanks! I'm not sure what that is. I meant something like this:

![Peek 2021-06-26 21-58](https://user-images.githubusercontent.com/924196/123524318-ae8a4580-d6c9-11eb-9810-9c95760a1e09.gif)

Reply:
> The param with only 2 methods is incorrect. I have a theory that this PR changes the autocomplete from 5 methods to 2. That cannot happen.

yep, you are right. it reduces to only 2.

Reply:
Not publicly. PHPStorm claim they "support generics", but they only mean e.g. autocomplete in `@template <here>` docblock, not actually what we talk about here. 

Here is 2 years old issue about it:
https://youtrack.jetbrains.com/issue/WI-47158

Upvote might help a bit